src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -1,9 +1,257 @@
 import React from 'react';
+import Cropper from 'react-cropper';
+import Dropzone from 'react-dropzone';
+import ErrorableFileInput from '../../common/components/form-elements/ErrorableFileInput';
+
+const FILE_TYPES = [
+  'png',
+  'tiff',
+  'tif',
+  'jpeg',
+  'jpg',
+  'bmp'
+];
+const MIN_SIZE = 350;
+
+function detectDragAndDrop() {
+  const div = document.createElement('div');
+  const supportsDragAndDrop = ('draggable' in div) || ('ondragstart' in div && 'ondrop' in div);
+  const iOS = !!navigator.userAgent.match('iPhone OS') || !!navigator.userAgent.match('iPad');
+  const dragAndDropDetected = supportsDragAndDrop && window.FileReader;
+  return dragAndDropDetected && !iOS;
+}
+
+function isValidFileType(file) {
+  return FILE_TYPES.some(type => file.name.toLowerCase().endsWith(type));
+}
+
+function isValidImageSize(file) {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.onload = () => {
+      const img = new Image();
+      img.src = reader.result;
+      img.onerror = (e) => reject(e);
+      img.onload = () => {
+        const isValidSize = img.width >= MIN_SIZE && img.height >= MIN_SIZE;
+        resolve(isValidSize);
+      };
+    };
+  });
+}
+
+function isValidImage(file) {
+  return new Promise((resolve, reject) => {
+    isValidImageSize(file)
+      .then(isValidSize => {
+        const isValidType = isValidFileType(file);
+        resolve(isValidSize && isValidType);
+      })
+      .catch((e) => reject(e));
+  });
+}
+
 
 export default class PhotoField extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      src: null,
+      cropResult: null,
+      done: false,
+      zoomValue: 2,
+      errorMessage: null
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const newView = !prevState.src || prevState.done;
+    const cropper = this.refs.cropper;
+    if (newView && cropper) {
+      setTimeout(() => {
+        const containerData = cropper.getContainerData();
+        const containerWidth = containerData.width;
+        const smallScreen = window.innerWidth < 768;
+        const cropBoxSize = smallScreen ? 240 : 300;
+        const cropBoxLeftOffset = (containerWidth - cropBoxSize) / 2;
+        const cropBoxData = {
+          left: cropBoxLeftOffset,
+          top: 0,
+          width: cropBoxSize,
+          height: cropBoxSize
+        };
+        cropper.setCropBoxData(cropBoxData);
+      }, 0);
+    }
+  }
+
+  onDone = () => {
+    this.setState({ src: null, done: true });
+  }
+
+  onChange = (files) => {
+    if (files && files[0]) {
+      const file = files[0];
+      const reader = new FileReader();
+      reader.readAsDataURL(file);
+      reader.onload = () => {
+        isValidImage(file)
```

Comment:
It seems like we're reading in the file twice. Once on line 97, then again in isValidImage.

Reply:
Good catch, @jbalboni, this has been changed.